pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BaseTableUpsertMetadataManager.java

```diff
@@ -55,20 +39,11 @@ public abstract class BaseTableUpsertMetadataManager implements TableUpsertMetad
   private static final Logger LOGGER = LoggerFactory.getLogger(BaseTableUpsertMetadataManager.class);
 
   protected String _tableNameWithType;
-  protected TableDataManager _tableDataManager;
-  protected HelixManager _helixManager;
-  protected ExecutorService _segmentPreloadExecutor;
```

Comment:
these are all breaking changes. Third party code may be extending this class and using these attributes

Reply:
those were added when adding the preloading logic but was made at table level. Those got reverted in this PR to move the preloading logic to table partition level for more flexibility and more explicit/clear locking logic.